.github/workflows/build_and_publish.yml

```diff
@@ -0,0 +1,60 @@
+---
+name: Tag new version, build and publish Docker image
+
+on:
+  workflow_run:
+    workflows:
+      - CI
+    branches:
+      - main
+    types:
+      - completed
+
+env:
+  IMAGE_NAME: sqlrunner
+
+jobs:
+  tag-new-version:
+    name: Tag new version
+    runs-on: ubuntu-latest
+    outputs:
+      new_version: ${{ steps.tag.outputs.new_version }} # no prefix, may be `undefined`
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # all history
+      - id: tag
+        uses: mathieudutour/github-tag-action@d745f2e74aaf1ee82e747b181f7a0967978abee0 # v6.0
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          create_annotated_tag: true
+          # FIXME
+          dry_run: true
+
+  build-and-publish-docker-image:
+    name: Build and publish Docker image
+    runs-on: ubuntu-latest
+    needs: tag-new-version
+    if: needs.tag-new-version.outputs.new_version
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set VERSION environment variable
+        run: echo VERSION=${{ needs.tag-new-version.outputs.new_version }} >> $GITHUB_ENV
+      - name: Write VERSION file
+        run: echo $VERSION > sqlrunner/VERSION
+      - name: Build image
```

Comment:
I think this should use the just command to build it, or else they drift.

Reply:
Note that our other projects use docker-compose to scaffold the building process, as it's declarative, and makes it easier to manage the build process.

If we later add a dev image to sqlrunner, like we do in other projects, it will help building that too.

Reply:
Thanks Simon. 1ee5a9c renames the just command; 133850c uses this command to build the production image.